### PR TITLE
Version of the Key

### PR DESCRIPTION
**Pull request by D-o-c**
Hi!

I have added the function version in the Key class to retrieve the version of the key directly (exactly the same idea of the fingerprint function).
 
:)
Best Regards
Mattia Pini

### CODE REVIEW

File: Misc/pgptime.cpp [hunk 1]
```diff
@@ -25,6 +25,13 @@ std::string show_time(time_t time){
     return date.str();
 }
 
+// write a time following strftime format
+std::string show_time_format(time_t time, const char* format = "%F %T", uint8_t limit = 80){
+        char buffer [limit];
```
Comment (calccrypto) [inline Misc/pgptime.cpp]: No variable length arrays please

File: Misc/pgptime.cpp [hunk 2]
```diff
@@ -25,6 +25,13 @@ std::string show_time(time_t time){
     return date.str();
 }
 
+// write a time following strftime format
+std::string show_time_format(time_t time, const char* format = "%F %T", uint8_t limit = 80){
+        char *buffer = new char(limit);
+        strftime (buffer, limit, format, localtime (&time));
+        return buffer;
```
Comment (calccrypto) [inline Misc/pgptime.cpp]: There be a memory leak

**Comment by D-o-c** [comment]
now should be ok

**Comment by calccrypto** [comment]
The `version` function is fine. I just don't see the need for `show_time_format`

**Comment by D-o-c** [comment]
I need that function for another project (I need the time in a specific format) and I thought that maybe it can be useful for somebody else.

**Comment by calccrypto** [comment]
Merged a972e4eccbf33907e348bdf7dfc310dab2b84e3f, ca9d8302861add445a2595fa10ed038b354f0def, and e619cbe081a56a5c0bf0c7da12045983cfe6cd23

**Pull request closed without merging**